examples/fluids/navierstokes.c

```diff
@@ -363,7 +357,7 @@ int main(int argc, char **argv) {
 
   // -- Operators
   CeedOperatorDestroy(&ceed_data->op_setup_vol);
-  CeedOperatorDestroy(&ceed_data->op_ics);
+  OperatorApplyContextDestroy(&ceed_data->op_ics_ctx);
   CeedOperatorDestroy(&user->op_rhs_vol);
   CeedOperatorDestroy(&user->op_ifunction_vol);
   CeedOperatorDestroy(&user->op_rhs);
```

Comment:
I had a internal debate about that, but couldn't come to an consensus. It comes down to two main points for me. 

1) Even as we continue to expand this out, it's still an example in the library repo. Exposing the guts in a clear way is a good chunk of it's purpose. I think abstractifying the more periphery features of the code is perfectly reasonable, but it could be argued that the core bits of the code, like this, might benefit from being left alone.

2) More of an issue for IFunction than RHS, but these have some edge cases to them which might warrant leaving them expanded. For example, I'm not sure how to keep the overlapping of `GlobalToLocal` communication for `Q` and `Q_dot` with a `ApplyCeedOperatorGlobalToLocal` call. We could keep it to `ApplyCeedOperatorLocalToLocal`, but then we have abstraction without as much benefit (as far as lines of code removed) while detracting from point 1.

Reply:
I tend to use the wrappers as much as possible since they make sure all the pieces fit together and help me not write some subtle bug.

1 isn't really compelling to me since we have the lower level examples and eventually we don't want anyone to have to write most of this 'glue' code themselves if they are using PETSc.

2 is a more interesting case to me. It makes me think we should merge this work with the operators we can wrap now and revisit in the future the  question of overlapping comm.

Reply:
Yeah, sounds good to me.

Reply:
I've now pushed up migrating RHS to the `ApplyCeedOperator*` wrappers. Other than that, the other direct uses of `CeedOperatorApply` are for quadrature point data that is only used as passive inputs for QFunctions.